### PR TITLE
Add 2 wizards with 1st stage and boilerplate for stage 2 and 3

### PR DESCRIPTION
**Pull request by paweljanicki**
LATEST UPDATE: (Already review in the PR https://github.com/PrimeDAO/prime-deals-dapp/pull/86)
- Add "make an offer wizard"
- Add boilerplate for stages "Proposal Lead" and "Primary DAO" and "PartnerDAO"
- Add some components shared across stages
- Passing data to stage components via route settings

UPDATE:
- First stage of the wizard
- stageWrapper component to reuse layout and basic styling between stages (no logic)
- Refactor and improvements to pstepper component

https://prime-deals-dapp-git-feature-wizard-stage-1-curvelabs.vercel.app/initiate/token-swap/open-proposal/stage1

ALREADY REVIEWED:
Two steps for selecting a deal type

Step 1: https://www.figma.com/file/hTLvqCEzki3WoG5uJbtynw/Prime-Deals?node-id=3566%3A70484
Step 2: https://www.figma.com/file/hTLvqCEzki3WoG5uJbtynw/Prime-Deals?node-id=3566%3A70438

I have:
- refactored code from the step one and made it a reusable component
- added e2e tests for selecting deal type
- connected the new "openDealWizard" (WIP) to the flow

### CODE REVIEW

File: cypress/integration/features/deals/choose-deal-type.feature [hunk 1]
```diff
@@ -1,9 +1,21 @@
...
+    When I click select on Token Swap card
+    Then I am redirected to Do you have a partner page
 
-    Scenario: Choose deal types
\ No newline at end of file
+  Scenario: I can see correct Token Swap types
+    Given I click select on Token Swap card
+    Then I can see Open Proposal and Partnered Deal
+    
+  Scenario: I can see select Open Proposal
```
Comment (hiaux0) [inline cypress/integration/features/deals/choose-deal-type.feature:18]:
```suggestion
  Scenario: Select Open Proposal
```

File: cypress/integration/features/deals/choose-deal-type.feature [hunk 2]
```diff
@@ -1,9 +1,21 @@
...
+    Given I click select on Token Swap card
+    And I click select Open Proposal
```
Comment (hiaux0) [inline cypress/integration/features/deals/choose-deal-type.feature:20]:
```suggestion
    And I select Open Proposal   // or "choose" instead "select"
```

⚠️  **This suggestions generally applies to all places you used "click"**

File: cypress/integration/features/deals/choose-deal-type.feature [hunk 3]
```diff
@@ -1,9 +1,21 @@
...
+    Then I am redirected to Open Proposal wizard
```
Comment (hiaux0) [inline cypress/integration/features/deals/choose-deal-type.feature:21]:
```suggestion
    Then I can view the Open Proposal wizard
```
**Reason:** An assumption is made about how the UI is done (could open in a modal as well?)

Note: I'm not 100% happy with my suggestion either

File: cypress/integration/tests/deals/choose-deal-type.e2e.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  cy.url().should("include", "token-swap")
+})
+
+Then("I can see Open Proposal and Partnered Deal", () => {
+  cy.contains("[data-test='initiate-deal-type-title']", "Open Proposal").should("be.visible")
+  cy.contains("[data-test='initiate-deal-type-title']", "Partnered Deal").should("be.visible")
+})
+
+And('I click select Open Proposal', () => {
+  cy.get("[data-test='button-initiate/token-swap/open-proposal/stage1']").click()
```
Comment (hiaux0) [inline cypress/integration/tests/deals/choose-deal-type.e2e.ts:30]: Where is this particular selector "button-initiate/token-swap/open-proposal/stage1"?

**Review by hiaux0 — COMMENTED** [review]
## Code
- Well done on writing these tests! 🎉 
- General structure is good
- Left some suggestions on formulation

- Skimmed through the rest of the code -> okay on higher level

**Review by dkent600 — CHANGES_REQUESTED** [review]
Can we add tests for the navbar links?

File: cypress/integration/features/deals/choose-deal-type.feature [hunk 5]
```diff
@@ -1,9 +1,21 @@
 Feature: Choose Deal type
-    Background:
-      Given I navigate to the Deals home page
-      And I navigate to the All Deals page
+  Background:
+    Given I navigate to the Deals home page
+    And I navigate to the initiate a deal page by clicking Initiate a Deal
```
[the remark this answers is not in the document]
Comment (paweljanicki) [reply cypress/integration/features/deals/choose-deal-type.feature:4]: Giving it more thought, I agree that describing the detail here doesn't make sense, as it's a background step. 
Probably there should be another feature that actually tests if the "Initiate a Deal" button exists and is clickable

File: cypress/integration/features/deals/choose-deal-type.feature [hunk 6]
```diff
@@ -1,9 +1,21 @@
...
+    
+  Scenario: I can see correct deal types
+    Then I can see Token Swap deal type
+    And I can see Joint Venture deal type
 
-    Scenario: Read about deal types
-      Then I can read about the deal types
+  Scenario: I can select Token Swap
+    When I click select on Token Swap card
+    Then I am redirected to Do you have a partner page
```
[the remark this answers is not in the document]
Comment (paweljanicki) [reply cypress/integration/features/deals/choose-deal-type.feature:12]: thumbsup

File: cypress/integration/features/deals/choose-deal-type.feature [hunk 7]
```diff
@@ -1,9 +1,21 @@
...
 
-    Scenario: Choose deal types
\ No newline at end of file
+  Scenario: I can see correct Token Swap types
```
[the remark this answers is not in the document]
Comment (paweljanicki) [reply cypress/integration/features/deals/choose-deal-type.feature:14]: That's a good point, I want to check user is present with correct options. And I want to check if he can successfully select one of the presented options. Should it be one scenario or two?

File: cypress/integration/tests/deals/choose-deal-type.e2e.ts (hunk 4 above)
Comment (paweljanicki) [reply cypress/integration/tests/deals/choose-deal-type.e2e.ts:30]: Screen Shot 2022-01-20 at 2 43 30 pm

File: cypress/integration/features/deals/choose-deal-type.feature (hunk 3 above)
Comment (paweljanicki) [reply cypress/integration/features/deals/choose-deal-type.feature:21]: Makes sense

**Comment by paweljanicki** [comment]
> Can we add tests for the navbar links?

@hiaux0 this is maybe something you are already working on? Otherwise maybe we create ticket for that and do in a separate branch

File: cypress/integration/features/deals/choose-deal-type.feature (hunk 5 above)
[the remark this answers is not in the document]
Comment (hiaux0) [reply cypress/integration/features/deals/choose-deal-type.feature]: > another feature that actually tests if the "Initiate a Deal" button exists and is clickable

I would say the step "    And I navigate to the initiate a deal page" already does this.
It is okay to have this button be implicitly tested like this.

This is the little difference between BDD and TDD.
In TDD you would have had one specific test, that covers button clicking.

File: cypress/integration/features/deals/choose-deal-type.feature (hunk 7 above)
[the remark this answers is not in the document]
Comment (hiaux0) [reply cypress/integration/features/deals/choose-deal-type.feature]: I'd go with merging.
We can leave it as two for now, but I'm confident we want to have just one Scenario for cases like this

File: cypress/integration/tests/deals/choose-deal-type.e2e.ts (hunk 4 above)
Comment (hiaux0) [reply cypress/integration/tests/deals/choose-deal-type.e2e.ts:30]: Can you link me to the place in the code?
I'm just wondering about the css selector having "/"

File: src/initiate/dealTypeSelector/dealTypeSelector.html [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    </div>
+    <div class="dealTypeBoxes">
+      <template repeat.for="box of boxes">
+        <div class="dealTypeBox">
+          <div data-test="initiate-deal-type-title" class="heading heading3 title">${box.name}</div>
+          <div class="subtext">
+            ${box.description}
+          </div>
+          <pbutton
+            data-test="button-${box.slug}"
```
Comment (paweljanicki) [inline src/initiate/dealTypeSelector/dealTypeSelector.html:20]: @hiaux0 data-test

File: cypress/integration/features/deals/choose-deal-type.feature (hunk 5 above)
[the remark this answers is not in the document]
Comment (paweljanicki) [reply cypress/integration/features/deals/choose-deal-type.feature]: Alright, that makes it simple

**Comment by hiaux0** [comment]
> navbar tests

@dkent600 
Do you mean the second row?:
Imo, not a blocker for this PR, especially, as only "Home" and "Initiate a Deal" have content.
For both, there are navigation tests already.

image

**Comment by dkent600** [comment]
> @dkent600 Do you mean the second row?: Imo, not a blocker for this PR, especially, as only "Home" and "Initiate a Deal" have content. For both, there are navigation tests already.

Yes, second row is the "submenu".  I was only asking about Initiate a Deal, and if it is already correctly handled and passing, then that satisfies my request, thanks

**Review by hiaux0 — COMMENTED** [review]
Good to go from my side ✔️

**Review by dkent600 — APPROVED** [review]

**Pull request merged**